java-lib/src/main/java/org/flinkextended/clink/util/ByteArrayDecoder.java

```diff
@@ -38,11 +38,18 @@
             @Override
             public byte[] read() throws IOException {
                 try {
-                    int expectedLen = inputViewStreamWrapper.readInt();
-                    byte[] bytes = new byte[expectedLen];
-                    int actualLen = inputViewStreamWrapper.read(bytes);
-                    Preconditions.checkArgument(expectedLen == actualLen);
-                    return bytes;
+                    int nRead;
```

Comment:
The `StreamFormat.Reader::read()` Java doc says:
```
Reads the next record. Returns {@code null} when the input has reached its end.
```

So the reader returned by `ByteArrayDecoder` should be able to return multiple model data (of type `byte[]`) if there are so many values in the given `inputStream`, which could be useful when we re-transmit multiple model data for online learning.

However, it seems that the reader returned here can only support returning at most one model data, because the implementation relies on the end-of-stream to detect end of the model data.

Reply:
You are right. I forgot this point. I'll add this back.

> If the only reader of the output data, which is the C++ OneHotEncoderModel::load, does not need to read byte length, would it be simpler not to output this byte length? Then we can also make the logic here consistent with the C++ saveMetaDataModelData(...).

Then as for the previous comment as above, I think we should still also record the length in C++ programs, because C++ and Java programs should read or write to files with the same format.

Reply:
We have discussed offline that protobuf should be able to record the length of protobuf data in the byte array, so we do not need to explicitly record the length. However, as its document pointed out

> If you want to write multiple messages to a single file or stream, it is up to you to keep track of where one message ends and the next begins. The Protocol Buffer wire format is not self-delimiting, so protocol buffer parsers cannot determine where a message ends on their own. The easiest way to solve this problem is to write the size of each message before you write the message itself. When you read the messages back in, you read the size, then read the bytes into a separate buffer, then parse from that buffer. (If you want to avoid copying bytes to a separate buffer, check out the CodedInputStream class (in both C++ and Java) which can be told to limit reads to a certain number of bytes.)
https://developers.google.com/protocol-buffers/docs/techniques

Protobuf does not have such ability, and recording the size on our own is its recommendation. Thus I believe it is reasonable to make this change in PR.